### PR TITLE
Refine debugging doc

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Part of docs revamp #7666

Just made some small changes and added link references.
Also added profiling since it's sort of part of debugging process in case of bottlenecks.
Are there any other flags we should add here?

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @rohitgr7

### CODE REVIEW

File: docs/source/common/debugging.rst [hunk 1]
```diff
@@ -20,9 +23,10 @@ The following are flags that make debugging much easier.
 ----------------
 
 fast_dev_run
-------------
-This flag runs a "unit test" by running n if set to ``n`` (int) else 1 if set to ``True`` training and validation batch(es).
-The point is to detect any bugs in the training/validation loop without having to wait for a full epoch to crash.
+============
+
+This flag runs a "unit test" by running ``N`` if set to ``N`` (int) else 1 if set to ``True`` training, validation, testing and predict batch(es)
```
Comment (Borda) [inline docs/source/common/debugging.rst:28]: shall we also mention that is the same as limiting training/valid batches?

**Review by Borda — APPROVED** [review]

File: docs/source/common/debugging.rst [hunk 2]
```diff
@@ -4,8 +4,11 @@
 
 .. _debugging:
 
+#########
 Debugging
-=========
+#########
+
+Lightning Trainer is empowered with a lot of debugging flags that can help you debug your :class:`~pytorch_lightning.core.lightning.LightningModule`.
```
Comment (awaelchli) [inline docs/source/common/debugging.rst:11]:
```suggestion
The Lightning Trainer is empowered with a lot of debugging flags that can help you debug your :class:`~pytorch_lightning.core.lightning.LightningModule`.
```

Comment (awaelchli) [reply docs/source/common/debugging.rst:11]: also word repetition debugging

File: docs/source/common/debugging.rst [hunk 3]
```diff
@@ -68,8 +75,8 @@ Monitor and log device stats during training with the :class:`~pytorch_lightning
 
 ----------------
 
-Make model overfit on subset of data
-------------------------------------
+Make Model overfit on Subset of Data
```
Comment (awaelchli) [inline docs/source/common/debugging.rst:78]:
```suggestion
Make Model Overfit on Subset of Data
```

File: docs/source/common/debugging.rst [hunk 4]
```diff
@@ -85,16 +92,18 @@ argument of :class:`~pytorch_lightning.trainer.trainer.Trainer`)
...
 
-With this flag, the train, val, and test sets will all be the same train set. We will also replace the sampler
+When using this flag, validation will be disabled. We will also replace the sampler
 in the training set to turn off shuffle for you.
 
 ----------------
 
-Print a summary of your LightningModule
----------------------------------------
+Print a Summary of your LightningModule
```
Comment (awaelchli) [inline docs/source/common/debugging.rst:100]:
```suggestion
Print a Summary of Your LightningModule
```

File: docs/source/common/debugging.rst [hunk 5]
```diff
@@ -152,3 +163,10 @@ argument of :class:`~pytorch_lightning.trainer.trainer.Trainer`)
 
     # DEFAULT
     trainer = Trainer(num_sanity_val_steps=2)
+
+----------------
+
+Profiling
+=========
+
+Checkout the :ref:`profiler` doc.
```
Comment (awaelchli) [inline docs/source/common/debugging.rst:172]: same

File: docs/source/common/debugging.rst [hunk 6]
```diff
@@ -85,16 +92,18 @@ argument of :class:`~pytorch_lightning.trainer.trainer.Trainer`)
...
 
 ----------------
 
-Print a summary of your LightningModule
----------------------------------------
+Print a Summary of your LightningModule
+=======================================
+
 Whenever the ``.fit()`` function gets called, the Trainer will print the weights summary for the LightningModule.
+If is enabled by default and can be turned-off using ``Trainer(enable_model_summary=False)``.
```
Comment (awaelchli) [inline docs/source/common/debugging.rst:104]:
```suggestion
If is enabled by default and can be turned off using ``Trainer(enable_model_summary=False)``.
```

Comment (awaelchli) [reply docs/source/common/debugging.rst:104]: I would mention this last, since from a debugging perspective you want to highlight the fact that you can set the depth. Mentioning how to disable it as the very first thing seems odd.
Also, would cause repetition of the default case in the next sentence

**Review by awaelchli — APPROVED** [review]
minor comments

File: docs/source/common/debugging.rst [hunk 7]
```diff
@@ -20,9 +23,10 @@ The following are flags that make debugging much easier.
 ----------------
 
 fast_dev_run
-------------
-This flag runs a "unit test" by running n if set to ``n`` (int) else 1 if set to ``True`` training and validation batch(es).
-The point is to detect any bugs in the training/validation loop without having to wait for a full epoch to crash.
+============
+
+This flag runs a "unit test" by running ``N`` if set to ``N`` (int) else 1 if set to ``True`` training, validation, testing and predict batch(es)
+for a single epoch. The point is to detect any bugs in the respective loop without having to wait for a complete loop to crash.
```
Comment (kaushikb11) [inline docs/source/common/debugging.rst:29]:
```suggestion
for a single epoch. The point is to have a dry run to detect any bugs in the respective loop without having to wait for a complete loop to crash.
```
It's a term in tech

File: docs/source/common/debugging.rst (hunk 6 above)
Comment (kaushikb11) [inline docs/source/common/debugging.rst:104]:
```suggestion
If enabled by default and can be turned off using ``Trainer(enable_model_summary=False)``.
```

**Review by kaushikb11 — APPROVED** [review]

**Review by ananthsub — COMMENTED** [review]
What about features like detect anomaly?
